toolchain/check/impl.cpp

```diff
@@ -296,7 +296,16 @@ auto CheckConstraintIsInterface(Context& context, SemIR::InstId impl_decl_id,
     return SemIR::SpecificInterface::None;
   }
 
-  auto identified_id = RequireIdentifiedFacetType(context, *facet_type);
+  auto identified_id = RequireIdentifiedFacetType(context, *facet_type, [&] {
+    CARBON_DIAGNOSTIC(ImplOfUnidentifiedFacetType, Error,
+                      "impl as facet type {0} that cannot be identified",
```

Comment:
Maybe `that` can be dropped? But really, the phrasing of "impl as facet type" is hard for me to parse because "impl as facet type" sounds like "(impl) as (facet type)", but on scrutiny I think it's intended as "(impl as) [uses a] (facet type {0})". Thus I try dropping "impl as".

So with that understanding, I might suggest something like:

```suggestion
                      "facet type {0} cannot be identified",
```

Or:

```suggestion
                      "facet type {0} cannot be identified for `impl as`",
```

Reply:
Done